### ISSUE TITLE
hash_match: specify input separately for cases and controls

### ISSUE THREAD

**Issue by thomas-rasmussen** [thomas-rasmussen/sas_macros#6]
Right now the input dataset needs to be a single cohort with a date, if any, a person becomes a case.

Would it make sense to implement the possibility to separate an input dataset for cases and potential controls separately? This might be a moe natural starting point for some people, and it would also make it possible to be more flexible about what patients we want to find controls for. 

Alternatively a dichotomous input variable could be specified that controls whether controls should be found or not? This approach seems more elegant, and if you had separate input datasets you could simply combine them and use this approach?

**Comment by thomas-rasmussen** [comment]
Maybe some feedback from other users is needed to know whether or not this makes sense to implement

**Comment by thomas-rasmussen** [comment]
The more I think about it, the more I like an alternative approach where you specify the controls and cases separately. It is more intuitive and it is more flexible than the current approach?

**Comment by divaDseidnA** [comment]
I have used hash_match macro in multiple projects, and even extracted elements from it for some projects. What I found was that its approach requiring a single input table helps the code to flow before and after the macro with grace. So even though, a multi table approach is a natural though, I found this current version useful beyond its own purpose.

**Comment by thomas-rasmussen** [comment]
Yeah I think so too, at least most of the time. I reopened this issue because I am irritated with the flow of my code in a current project where using the macro feels clunky. Basically I have two (partially overlapping) populations, where I want to find controls from the same source population. I'm merging index dates to the source population and matching, one population at the time, then combining the matched populations afterwards. This does not feel great. My thought was, that if the cases and controls are specified separately, then I could simply stack my two populations, and do everything at once (I'm matching with replacement). Generally speaking, it would also always be trivial to modify the input dataset into cases and controls with a single datastep, so changing the macro would not really affect how you set up your code to use it, but it might make it more intuitive/flexible in some scenarios?

Writing this I also just realized, that I can "fix" my issue by making a copy of my sourcepopulation, merge one population to each copy, then add an indicator variable to the exact matching variables. But I think the code would be much more readable, if I simply had to stack my populations and input that as cases, and then have my source population as controls.
